### DISCUSSION TITLE
getting error when Building from source srsRan Project Makefile:1446:all Error 2

### DISCUSSION THREAD

**Discussion by mrsaad381** [srsran/srsRAN_Project#659] [General Help]
Hi, i am facing error from two to three days itrying all possible solution 
error is releated to lib/srcvec/ directory
in this directory we have all cpp files for vectors and error belong to its types of vector or length
Here is descripition of error  in details copy paste from terminal
[  6%] Built target srsgnb_app_f1u_connector
[  6%] Building CXX object lib/srsvec/CMakeFiles/srsvec.dir/accumulate.cpp.o
[  6%] Building CXX object lib/fapi/loggers/CMakeFiles/srsran_fapi_loggers.dir/logging_decorator_factories.cpp.o
[  6%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/cause/e1ap_cause_converters.cpp.o
[  6%] Building CXX object lib/support/network/CMakeFiles/srsran_network.dir/sockets.cpp.o
In file included from /home/vboxuser/srsRAN_Project/lib/srsvec/accumulate.cpp:23:
/home/vboxuser/srsRAN_Project/lib/srsvec/simd.h: In function 'srsran::simd_cf_t srsran::srsran_simd_cf_interleave_low(srsran::simd_cf_t, srsran::simd_cf_t)':
/home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:1212:40: error: cannot convert 'srsran::simd_f_t' {aka '__m128'} to '__m256'
 1212 |   __m256 temp_a0 = _mm256_permute_ps(a.re, 0b11011000);
      |                                      ~~^~
      |                                        |
      |                                        srsran::simd_f_t {aka __m128}
In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/immintrin.h:43,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:36,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/accumulate.cpp:23:
/usr/lib/gcc/x86_64-linux-gnu/11/include/avxintrin.h:668:27: note:   initializing argument 1 of '__m256 _mm256_permute_ps(__m256, int)'
  668 | _mm256_permute_ps (__m256 __X, const int __C)
      |                    ~~~~~~~^~~
In file included from /home/vboxuser/srsRAN_Project/lib/srsvec/accumulate.cpp:23:
/home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:1213:40: error: cannot convert 'srsran::simd_f_t' {aka '__m128'} to '__m256'
 1213 |   __m256 temp_a1 = _mm256_permute_ps(a.im, 0b11011000);
      |                                      ~~^~
      |                                        |
      |                                        srsran::simd_f_t {aka __m128}
In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/immintrin.h:43,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:36,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/accumulate.cpp:23:
/usr/lib/gcc/x86_64-linux-gnu/11/include/avxintrin.h:668:27: note:   initializing argument 1 of '__m256 _mm256_permute_ps(__m256, int)'
  668 | _mm256_permute_ps (__m256 __X, const int __C)
      |                    ~~~~~~~^~~
In file included from /home/vboxuser/srsRAN_Project/lib/srsvec/accumulate.cpp:23:
/home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:1217:40: error: cannot convert 'srsran::simd_f_t' {aka '__m128'} to '__m256'
 1217 |   __m256 temp_b0 = _mm256_permute_ps(b.re, 0b11011000);
      |                                      ~~^~
      |                                        |
      |                                        srsran::simd_f_t {aka __m128}
In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/immintrin.h:43,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:36,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/accumulate.cpp:23:
/usr/lib/gcc/x86_64-linux-gnu/11/include/avxintrin.h:668:27: note:   initializing argument 1 of '__m256 _mm256_permute_ps(__m256, int)'
  668 | _mm256_permute_ps (__m256 __X, const int __C)
      |                    ~~~~~~~^~~
In file included from /home/vboxuser/srsRAN_Project/lib/srsvec/accumulate.cpp:23:
/home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:1218:40: error: cannot convert 'srsran::simd_f_t' {aka '__m128'} to '__m256'
 1218 |   __m256 temp_b1 = _mm256_permute_ps(b.im, 0b11011000);
      |                                      ~~^~
      |                                        |
      |                                        srsran::simd_f_t {aka __m128}
In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/immintrin.h:43,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:36,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/accumulate.cpp:23:
/usr/lib/gcc/x86_64-linux-gnu/11/include/avxintrin.h:668:27: note:   initializing argument 1 of '__m256 _mm256_permute_ps(__m256, int)'
  668 | _mm256_permute_ps (__m256 __X, const int __C)
      |                    ~~~~~~~^~~
In file included from /home/vboxuser/srsRAN_Project/lib/srsvec/accumulate.cpp:23:
/home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:1234:34: error: cannot convert '__m256' to 'srsran::simd_f_t' {aka '__m128'} in assignment
 1234 |   ret.re     = _mm256_unpacklo_ps(in1, in2);
      |                ~~~~~~~~~~~~~~~~~~^~~~~~~~~~
      |                                  |
      |                                  __m256
/home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:1235:34: error: cannot convert '__m256' to 'srsran::simd_f_t' {aka '__m128'} in assignment
 1235 |   ret.im     = _mm256_unpackhi_ps(in1, in2);
      |                ~~~~~~~~~~~~~~~~~~^~~~~~~~~~
      |                                  |
      |                                  __m256
/home/vboxuser/srsRAN_Project/lib/srsvec/simd.h: In function 'srsran::simd_cf_t srsran::srsran_simd_cf_interleave_high(srsran::simd_cf_t, srsran::simd_cf_t)':
/home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:1269:40: error: cannot convert 'srsran::simd_f_t' {aka '__m128'} to '__m256'
 1269 |   __m256 temp_a0 = _mm256_permute_ps(a.re, 0b11011000);
      |                                      ~~^~
      |                                        |
      |                                        srsran::simd_f_t {aka __m128}
In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/immintrin.h:43,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:36,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/accumulate.cpp:23:
/usr/lib/gcc/x86_64-linux-gnu/11/include/avxintrin.h:668:27: note:   initializing argument 1 of '__m256 _mm256_permute_ps(__m256, int)'
  668 | _mm256_permute_ps (__m256 __X, const int __C)
      |                    ~~~~~~~^~~
In file included from /home/vboxuser/srsRAN_Project/lib/srsvec/accumulate.cpp:23:
/home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:1270:40: error: cannot convert 'srsran::simd_f_t' {aka '__m128'} to '__m256'
 1270 |   __m256 temp_a1 = _mm256_permute_ps(a.im, 0b11011000);
      |                                      ~~^~
      |                                        |
      |                                        srsran::simd_f_t {aka __m128}
In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/immintrin.h:43,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:36,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/accumulate.cpp:23:
/usr/lib/gcc/x86_64-linux-gnu/11/include/avxintrin.h:668:27: note:   initializing argument 1 of '__m256 _mm256_permute_ps(__m256, int)'
  668 | _mm256_permute_ps (__m256 __X, const int __C)
      |                    ~~~~~~~^~~
In file included from /home/vboxuser/srsRAN_Project/lib/srsvec/accumulate.cpp:23:
/home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:1274:40: error: cannot convert 'srsran::simd_f_t' {aka '__m128'} to '__m256'
 1274 |   __m256 temp_b0 = _mm256_permute_ps(b.re, 0b11011000);
      |                                      ~~^~
      |                                        |
      |                                        srsran::simd_f_t {aka __m128}
In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/immintrin.h:43,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:36,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/accumulate.cpp:23:
/usr/lib/gcc/x86_64-linux-gnu/11/include/avxintrin.h:668:27: note:   initializing argument 1 of '__m256 _mm256_permute_ps(__m256, int)'
  668 | _mm256_permute_ps (__m256 __X, const int __C)
      |                    ~~~~~~~^~~
In file included from /home/vboxuser/srsRAN_Project/lib/srsvec/accumulate.cpp:23:
/home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:1275:40: error: cannot convert 'srsran::simd_f_t' {aka '__m128'} to '__m256'
 1275 |   __m256 temp_b1 = _mm256_permute_ps(b.im, 0b11011000);
      |                                      ~~^~
      |                                        |
      |                                        srsran::simd_f_t {aka __m128}
In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/immintrin.h:43,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:36,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/accumulate.cpp:23:
/usr/lib/gcc/x86_64-linux-gnu/11/include/avxintrin.h:668:27: note:   initializing argument 1 of '__m256 _mm256_permute_ps(__m256, int)'
  668 | _mm256_permute_ps (__m256 __X, const int __C)
      |                    ~~~~~~~^~~
In file included from /home/vboxuser/srsRAN_Project/lib/srsvec/accumulate.cpp:23:
/home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:1291:34: error: cannot convert '__m256' to 'srsran::simd_f_t' {aka '__m128'} in assignment
 1291 |   ret.re     = _mm256_unpacklo_ps(in1, in2);
      |                ~~~~~~~~~~~~~~~~~~^~~~~~~~~~
      |                                  |
      |                                  __m256
/home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:1292:34: error: cannot convert '__m256' to 'srsran::simd_f_t' {aka '__m128'} in assignment
 1292 |   ret.im     = _mm256_unpackhi_ps(in1, in2);
      |                ~~~~~~~~~~~~~~~~~~^~~~~~~~~~
      |                                  |
      |                                  __m256
/home/vboxuser/srsRAN_Project/lib/srsvec/accumulate.cpp: In function 'float accumulate_f_simd(const float*, unsigned int)':
/home/vboxuser/srsRAN_Project/lib/srsvec/accumulate.cpp:44:14: error: 'srsran_simd_f_sum' was not declared in this scope
   44 |     result = srsran_simd_f_sum(simd_result);
      |              ^~~~~~~~~~~~~~~~~
[  6%] Building CXX object lib/srsvec/CMakeFiles/srsvec.dir/add.cpp.o
/home/vboxuser/srsRAN_Project/lib/srsvec/accumulate.cpp: At global scope:
/home/vboxuser/srsRAN_Project/lib/srsvec/accumulate.cpp:30:14: error: 'float accumulate_f_simd(const float*, unsigned int)' defined but not used [-Werror=unused-function]
   30 | static float accumulate_f_simd(const float* x, unsigned len)
      |              ^~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [lib/srsvec/CMakeFiles/srsvec.dir/build.make:76: lib/srsvec/CMakeFiles/srsvec.dir/accumulate.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[  6%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/csi_report/csi_report_config_helpers.cpp.o
[  6%] Building CXX object lib/fapi/loggers/CMakeFiles/srsran_fapi_loggers.dir/logging_slot_data_notifier_decorator.cpp.o
[  6%] Building CXX object lib/fapi/validators/CMakeFiles/srsran_fapi_validators.dir/dl_pdsch_pdu.cpp.o
[  7%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/csi_report/csi_report_on_pucch_helpers.cpp.o
[  7%] Building CXX object lib/fapi/loggers/CMakeFiles/srsran_fapi_loggers.dir/logging_slot_error_notifier_decorator.cpp.o
In file included from /home/vboxuser/srsRAN_Project/lib/srsvec/add.cpp:24:
/home/vboxuser/srsRAN_Project/lib/srsvec/simd.h: In function 'srsran::simd_cf_t srsran::srsran_simd_cf_interleave_low(srsran::simd_cf_t, srsran::simd_cf_t)':
/home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:1212:40: error: cannot convert 'srsran::simd_f_t' {aka '__m128'} to '__m256'
 1212 |   __m256 temp_a0 = _mm256_permute_ps(a.re, 0b11011000);
      |                                      ~~^~
      |                                        |
      |                                        srsran::simd_f_t {aka __m128}
In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/immintrin.h:43,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:36,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/add.cpp:24:
/usr/lib/gcc/x86_64-linux-gnu/11/include/avxintrin.h:668:27: note:   initializing argument 1 of '__m256 _mm256_permute_ps(__m256, int)'
  668 | _mm256_permute_ps (__m256 __X, const int __C)
      |                    ~~~~~~~^~~
In file included from /home/vboxuser/srsRAN_Project/lib/srsvec/add.cpp:24:
/home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:1213:40: error: cannot convert 'srsran::simd_f_t' {aka '__m128'} to '__m256'
 1213 |   __m256 temp_a1 = _mm256_permute_ps(a.im, 0b11011000);
      |                                      ~~^~
      |                                        |
      |                                        srsran::simd_f_t {aka __m128}
In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/immintrin.h:43,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:36,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/add.cpp:24:
/usr/lib/gcc/x86_64-linux-gnu/11/include/avxintrin.h:668:27: note:   initializing argument 1 of '__m256 _mm256_permute_ps(__m256, int)'
  668 | _mm256_permute_ps (__m256 __X, const int __C)
      |                    ~~~~~~~^~~
In file included from /home/vboxuser/srsRAN_Project/lib/srsvec/add.cpp:24:
/home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:1217:40: error: cannot convert 'srsran::simd_f_t' {aka '__m128'} to '__m256'
 1217 |   __m256 temp_b0 = _mm256_permute_ps(b.re, 0b11011000);
      |                                      ~~^~
      |                                        |
      |                                        srsran::simd_f_t {aka __m128}
In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/immintrin.h:43,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:36,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/add.cpp:24:
/usr/lib/gcc/x86_64-linux-gnu/11/include/avxintrin.h:668:27: note:   initializing argument 1 of '__m256 _mm256_permute_ps(__m256, int)'
  668 | _mm256_permute_ps (__m256 __X, const int __C)
      |                    ~~~~~~~^~~
In file included from /home/vboxuser/srsRAN_Project/lib/srsvec/add.cpp:24:
/home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:1218:40: error: cannot convert 'srsran::simd_f_t' {aka '__m128'} to '__m256'
 1218 |   __m256 temp_b1 = _mm256_permute_ps(b.im, 0b11011000);
      |                                      ~~^~
      |                                        |
      |                                        srsran::simd_f_t {aka __m128}
In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/immintrin.h:43,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:36,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/add.cpp:24:
/usr/lib/gcc/x86_64-linux-gnu/11/include/avxintrin.h:668:27: note:   initializing argument 1 of '__m256 _mm256_permute_ps(__m256, int)'
  668 | _mm256_permute_ps (__m256 __X, const int __C)
      |                    ~~~~~~~^~~
In file included from /home/vboxuser/srsRAN_Project/lib/srsvec/add.cpp:24:
/home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:1234:34: error: cannot convert '__m256' to 'srsran::simd_f_t' {aka '__m128'} in assignment
 1234 |   ret.re     = _mm256_unpacklo_ps(in1, in2);
      |                ~~~~~~~~~~~~~~~~~~^~~~~~~~~~
      |                                  |
      |                                  __m256
/home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:1235:34: error: cannot convert '__m256' to 'srsran::simd_f_t' {aka '__m128'} in assignment
 1235 |   ret.im     = _mm256_unpackhi_ps(in1, in2);
      |                ~~~~~~~~~~~~~~~~~~^~~~~~~~~~
      |                                  |
      |                                  __m256
/home/vboxuser/srsRAN_Project/lib/srsvec/simd.h: In function 'srsran::simd_cf_t srsran::srsran_simd_cf_interleave_high(srsran::simd_cf_t, srsran::simd_cf_t)':
/home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:1269:40: error: cannot convert 'srsran::simd_f_t' {aka '__m128'} to '__m256'
 1269 |   __m256 temp_a0 = _mm256_permute_ps(a.re, 0b11011000);
      |                                      ~~^~
      |                                        |
      |                                        srsran::simd_f_t {aka __m128}
In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/immintrin.h:43,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:36,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/add.cpp:24:
/usr/lib/gcc/x86_64-linux-gnu/11/include/avxintrin.h:668:27: note:   initializing argument 1 of '__m256 _mm256_permute_ps(__m256, int)'
  668 | _mm256_permute_ps (__m256 __X, const int __C)
      |                    ~~~~~~~^~~
In file included from /home/vboxuser/srsRAN_Project/lib/srsvec/add.cpp:24:
/home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:1270:40: error: cannot convert 'srsran::simd_f_t' {aka '__m128'} to '__m256'
 1270 |   __m256 temp_a1 = _mm256_permute_ps(a.im, 0b11011000);
      |                                      ~~^~
      |                                        |
      |                                        srsran::simd_f_t {aka __m128}
In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/immintrin.h:43,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:36,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/add.cpp:24:
/usr/lib/gcc/x86_64-linux-gnu/11/include/avxintrin.h:668:27: note:   initializing argument 1 of '__m256 _mm256_permute_ps(__m256, int)'
  668 | _mm256_permute_ps (__m256 __X, const int __C)
      |                    ~~~~~~~^~~
In file included from /home/vboxuser/srsRAN_Project/lib/srsvec/add.cpp:24:
/home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:1274:40: error: cannot convert 'srsran::simd_f_t' {aka '__m128'} to '__m256'
 1274 |   __m256 temp_b0 = _mm256_permute_ps(b.re, 0b11011000);
      |                                      ~~^~
      |                                        |
      |                                        srsran::simd_f_t {aka __m128}
In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/immintrin.h:43,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:36,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/add.cpp:24:
/usr/lib/gcc/x86_64-linux-gnu/11/include/avxintrin.h:668:27: note:   initializing argument 1 of '__m256 _mm256_permute_ps(__m256, int)'
  668 | _mm256_permute_ps (__m256 __X, const int __C)
      |                    ~~~~~~~^~~
In file included from /home/vboxuser/srsRAN_Project/lib/srsvec/add.cpp:24:
/home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:1275:40: error: cannot convert 'srsran::simd_f_t' {aka '__m128'} to '__m256'
 1275 |   __m256 temp_b1 = _mm256_permute_ps(b.im, 0b11011000);
      |                                      ~~^~
      |                                        |
      |                                        srsran::simd_f_t {aka __m128}
In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/immintrin.h:43,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:36,
                 from /home/vboxuser/srsRAN_Project/lib/srsvec/add.cpp:24:
/usr/lib/gcc/x86_64-linux-gnu/11/include/avxintrin.h:668:27: note:   initializing argument 1 of '__m256 _mm256_permute_ps(__m256, int)'
  668 | _mm256_permute_ps (__m256 __X, const int __C)
      |                    ~~~~~~~^~~
In file included from /home/vboxuser/srsRAN_Project/lib/srsvec/add.cpp:24:
/home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:1291:34: error: cannot convert '__m256' to 'srsran::simd_f_t' {aka '__m128'} in assignment
 1291 |   ret.re     = _mm256_unpacklo_ps(in1, in2);
      |                ~~~~~~~~~~~~~~~~~~^~~~~~~~~~
      |                                  |
      |                                  __m256
/home/vboxuser/srsRAN_Project/lib/srsvec/simd.h:1292:34: error: cannot convert '__m256' to 'srsran::simd_f_t' {aka '__m128'} in assignment
 1292 |   ret.im     = _mm256_unpackhi_ps(in1, in2);
      |                ~~~~~~~~~~~~~~~~~~^~~~~~~~~~
      |                                  |
      |                                  __m256
[  7%] Linking CXX static library libsrsran_network.a
make[2]: *** [lib/srsvec/CMakeFiles/srsvec.dir/build.make:90: lib/srsvec/CMakeFiles/srsvec.dir/add.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:10833: lib/srsvec/CMakeFiles/srsvec.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[  7%] Building CXX object lib/fapi/loggers/CMakeFiles/srsran_fapi_loggers.dir/logging_slot_gateway_decorator.cpp.o
[  7%] Built target srsran_network
[  7%] Building CXX object lib/fapi/loggers/CMakeFiles/srsran_fapi_loggers.dir/logging_slot_time_notifier_decorator.cpp.o
[  7%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/csi_report/csi_report_on_pusch_helpers.cpp.o
[  7%] Building CXX object lib/fapi/validators/CMakeFiles/srsran_fapi_validators.dir/dl_ssb_pdu.cpp.o
[  7%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/csi_report/csi_report_on_puxch_helpers.cpp.o
[  7%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/csi_rs/csi_rs_pattern.cpp.o
[  7%] Linking CXX static library libsrsran_fapi_loggers.a
[  7%] Built target srsran_fapi_loggers
[  8%] Linking CXX static library libsrsran_split_7_2_app_unit_helpers.a
[  8%] Building CXX object lib/fapi/validators/CMakeFiles/srsran_fapi_validators.dir/message_validators.cpp.o
[  8%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/csi_rs/csi_rs_config_helpers.cpp.o
[  8%] Built target srsran_split_7_2_app_unit_helpers
[  8%] Building CXX object lib/fapi/validators/CMakeFiles/srsran_fapi_validators.dir/tx_precoding_and_beamforming_pdu.cpp.o
[  8%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/csi_rs/frequency_allocation_type.cpp.o
[  8%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/pdcch/cce_to_prb_mapping.cpp.o
[  8%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/pdcch/dci_packing.cpp.o
[  8%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/pdcch/pdcch_candidates.cpp.o
[  8%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/pdcch/pdcch_type0_css_coreset_config.cpp.o
[  8%] Linking CXX static library libsrsran_split_8_app_unit_helpers.a
[  8%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/pdcch/pdcch_type0_css_occasions.cpp.o
[  8%] Building CXX object lib/fapi/validators/CMakeFiles/srsran_fapi_validators.dir/uci_pdus.cpp.o
[  8%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/pdcch/search_space.cpp.o
[  8%] Built target srsran_split_8_app_unit_helpers
[  8%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/pdsch/dlsch_info.cpp.o
[  8%] Building CXX object lib/fapi/validators/CMakeFiles/srsran_fapi_validators.dir/ul_prach_pdu.cpp.o
[  8%] Building CXX object lib/fapi/validators/CMakeFiles/srsran_fapi_validators.dir/ul_pucch_pdu.cpp.o
[  8%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/pdsch/pdsch_mcs.cpp.o
[  8%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/pdsch/pdsch_antenna_port_mapping.cpp.o
[  8%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/prach/prach_configuration.cpp.o
[  8%] Building CXX object lib/fapi/validators/CMakeFiles/srsran_fapi_validators.dir/ul_pusch_pdu.cpp.o
[  8%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/prach/prach_cyclic_shifts.cpp.o
[  8%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/prach/prach_preamble_information.cpp.o
[  9%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/prach/prach_frequency_mapping.cpp.o
[  9%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/prach/prach_helper.cpp.o
[  9%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/precoding/precoding_codebooks.cpp.o
[  9%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/pusch/pusch_mcs.cpp.o
[  9%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/pusch/pusch_tpmi_select.cpp.o
[  9%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/pusch/ulsch_info.cpp.o
[  9%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/pusch/pusch_uci_beta_offset.cpp.o
[  9%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/qos/five_qi_qos_mapping.cpp.o
[  9%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/uci/uci_part2_size_calculator.cpp.o
[  9%] Linking CXX static library libsrsran_fapi_validators.a
[  9%] Built target srsran_fapi_validators
[  9%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/band_helper.cpp.o
[  9%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/resource_allocation/resource_allocation_frequency.cpp.o
[  9%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/sch/tbs_calculator.cpp.o
[  9%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/sch/sch_segmentation.cpp.o
[  9%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/srs/srs_bandwidth_configuration.cpp.o
[  9%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/srs/srs_information.cpp.o
[  9%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/ssb_freq_position_generator.cpp.o
[  9%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/ssb_gscn.cpp.o
[ 10%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/ssb_mapping.cpp.o
[ 10%] Building CXX object lib/ran/CMakeFiles/srsran_ran.dir/tdd_ul_dl_config.cpp.o
[ 10%] Linking CXX static library libsrsran_ran.a
[ 10%] Built target srsran_ran
make: *** [Makefile:146: all] Error 2

![error1](https://github.com/srsran/srsRAN_Project/assets/115063859/e3b8e015-52ce-4589-ad6c-9c21d288b031)

**Comment by pgawlowicz** [comment]
do you build the project inside a virtual machine?

**Reply by mrsaad381** [reply]
yes i build it in a virtual machine

**Reply by pgawlowicz** [reply]
please check this answer: https://github.com/srsran/srsRAN_Project/discussions/684#discussioncomment-9838721
